.github/workflows/ci.yml

```diff
@@ -0,0 +1,103 @@
+name: Pipeline
+on: [push]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Start timer
+        id: start_timer
+        run: |
+         echo "START_TIME=$(date +%s)" >> $GITHUB_ENV
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18'
+      - name: Install dependencies
+        run: cd angular/ && npm install
+      - name: Run tests
+        run: cd angular/ && npm run coverage
+
+
+      - name: Login to GitHub Container Registry
+        if: always()
+        uses: docker/login-action@v1
+        with:
+          registry: ${{env.REGISTRY}}
+          username: ${{github.actor}}
+          password: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Extract metadata (tags, labels) for Docker
+        if: always()
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+
+      - name: Build and push Docker image
+        if: always()
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Stop timer
+        if: always()
+        id: stop_timer
+        run: |
+          echo "STOP_TIME=$(date +%s)" >> $GITHUB_ENV
+
+      - name: Check environment variables
+        if: always()
+        run: |
+          START_TIME=$(echo ${{ env.START_TIME }})
+          STOP_TIME=$(echo ${{ env.STOP_TIME }})
+          EXECUTION_TIME=$((STOP_TIME - START_TIME))
+          HOURS=$((EXECUTION_TIME / 3600))
+          MINUTES=$((EXECUTION_TIME % 3600 / 60))
+          SECONDS=$((EXECUTION_TIME % 60))
+          echo "EXECUTION_TIME_HOURS=$HOURS" >> $GITHUB_ENV
+          echo "EXECUTION_TIME_MINUTES=$MINUTES" >> $GITHUB_ENV
+          echo "EXECUTION_TIME_SECONDS=$SECONDS" >> $GITHUB_ENV
+          echo "Execution time: $HOURS hours $MINUTES minutes $SECONDS seconds"
```

Comment:
lets refactor timing calculation this into using `gh cli`, https://cli.github.com/
your code will not need start/stop time recorded vars and your pipeline would become nicer.

Reply:
Don't see that it is fixed